File: algo-linear-search/python/linear_search.py
array_to_search_through = list(range(1, 1001))

def linear_search(value_to_find, array_to_search_through): 
    return next((number for number in range(len(array_to_search_through)) if array_to_search_through[number] == value_to_find), None) 

def global_linear_search(value, array_to_search):
    output_array = [number for number in range(len(array_to_search)) if array_to_search[number] == value]
    print(output_array)

File: algo-linear-search/js/linearSearch.js
const arrayToSearchThrough = [];
for (let i = 1; i <= 1000; i++) {
    arrayToSearchThrough.push(i);
}

exports.linearSearch = function(valueToFind, arrayToSearchThrough) {
    for (let i = 0; i < arrayToSearchThrough.length; i++) {
        if (valueToFind === arrayToSearchThrough[i]) {
            return i;
        }
    }
    return undefined;
}

exports.linearSearchGlobal = function(valueToFind, arrayToSearchThrough){
    let valueToFindIndices = [];
    for(let i = 0; i < arrayToSearchThrough.length; i++){
        if(valueToFind === arrayToSearchThrough[i]){
            valueToFindIndices.push(i)
        }
    }
    if (valueToFindIndices.length != 0) {
        return valueToFindIndices;
    } else{
        return undefined;
    }
}